grpc-gcp/src/main/java/com/google/cloud/grpc/GcpManagedChannel.java

```diff
@@ -1189,6 +1459,36 @@ private ChannelRef tryCreateNewChannel() {
     return null;
   }
 
+  private void maybeDynamicUpscale() {
+    if (!isDynamicScalingEnabled || channelRefs.size() >= maxSize) {
+      return;
+    }
+
+    if ((totalActiveStreams.get() / channelRefs.size()) >= maxRpcPerChannel) {
+      dynamicUpscale();
+    }
+  }
+
+  private synchronized void dynamicUpscale() {
+    if (!isDynamicScalingEnabled || channelRefs.size() >= maxSize) {
+      return;
+    }
+
+    if ((totalActiveStreams.get() / channelRefs.size()) >= maxRpcPerChannel) {
+      createNewChannel();
+      scaleUpCount++;
+    }
+  }
+
+  private boolean shouldScaleUp(int minStreams, int totalStreams) {
+    if (channelRefs.size() >= maxSize) {
+      // Pool is full.
+      return false;
+    }
+
+    return !isDynamicScalingEnabled && minStreams >= maxConcurrentStreamsLowWatermark;
```

Comment:
If `isDynamicScalingEnabled` is true, i.e., enabled, then `shouldScaleUp` will always return false. Is this expected?

Reply:
I see. Then can you rename the variable? It is a bit confusing when `isDynamicScalingEnabled` is turned on but `shouldScaleUp` is disabled.

Reply:
I am open to suggestions. I can't come up with a better name because it is used in other places too and answers the question "Is dynamic scaling enabled?"

Reply:
OK then let's add a comment for `shouldScaleUp`, which is for "pre-dynamic channel scaling".